crates/biome_service/src/file_handlers/javascript.rs

```diff
@@ -659,13 +660,24 @@ pub(crate) fn fix_all(params: FixAllParams) -> Result<FixFileResult, WorkspaceEr
                 }
 
                 for action in signal.actions() {
-                    // suppression actions should not be part of the fixes (safe or suggested)
-                    if action.is_suppression() {
-                        continue;
-                    }
-
                     match params.fix_file_mode {
+                        FixFileMode::ApplySuppressions => {
+                            if action.is_suppression()
+                                && action.applicability == Applicability::Always
+                            // TODO: This doesn't seem correct. I'm having to set `"nursery": { "all": false }` in my configuration
```

Comment:
This is problematic, though I can't sort out why this is. @ematipico let me know that [nursery rules are enabled in debug builds](https://github.com/biomejs/biome/pull/4008#discussion_r1782460430) but I'm seeing nursery rules get ignores written with a release build. 🤔